subsys/usb/device/usb_transfer.c

```diff
@@ -251,7 +251,8 @@ int usb_transfer(uint8_t ep, uint8_t *data, size_t dlen, unsigned int flags,
 
 	if (flags & USB_TRANS_WRITE) {
 		/* start writing first chunk */
-		k_work_submit_to_queue(&USB_WORK_Q, &trans->work);
+		ret = k_work_submit_to_queue(&USB_WORK_Q, &trans->work);
+		ret = MAX(0, ret);
```

Comment:
I often mix min() an max(): `MAX(0, ret)` would take the greatest value between `0` and `ret`, so if `ret` is negative (i.e. error), and only pick 0, 1 or 2 as described here:
https://docs.zephyrproject.org/latest/kernel/services/threads/workqueue.html#c.k_work_submit_to_queue

Is that what is intended?
Thanks

Reply:
Thanks for the feedback. 
You are totally right, this is an error. Unsurprisingly the work queue never caused an issue in tests.
Fix incoming.